Remove margin if no input text

lindseydew opened the pull request:
## What does this change?
This takes away the white space to the left of the Sign Up button in the case where the input text is hidden.

## Why?
It was [picked up](https://github.com/guardian/dotcom-rendering/pull/10198#issuecomment-1892427078) on the [PR that introduced it](https://github.com/guardian/dotcom-rendering/pull/10198)  

## Screenshots

| Before      | After      |
| ----------- | ---------- |
| ![daily](https://github.com/guardian/dotcom-rendering/assets/1202622/addcee9e-bbbe-4847-a34e-717c4e710004) | ![email-present](https://github.com/guardian/dotcom-rendering/assets/1202622/45d2fb65-0ce5-402d-927c-e41f2dde33da)||

The component when the input is present:
![no-email](https://github.com/guardian/dotcom-rendering/assets/1202622/300b368c-fe21-4dc7-9c91-660eee69dfac)


[before]: https://example.com/before.png
[after]: https://example.com/after.png

<!--
You can add extra rows by repeating the last row in the table and then using new unique labels. E.g.

| ![before2][] | ![after2][] |

You can then reference the labels and map them to corresponding links.

[before2]: https://example.com/before2.png
[after2]: https://example.com/after2.png
-->

<!--
## Running Chromatic

In order to run Chromatic as part of the CI checks, you will need to add the `run_chromatic` label to your PR. Once the label is added Chromatic will run on every push.

Please only add this once you are ready to check for visual regressions, our intention here is to reduce the amount of time Chromatic is run without being looked at.
-->

<!--
## Unexplained Chromatic diffs

We use Chromatic for visual regression testing on our Storybook stories. It's
generally pretty good, but it sometimes gives 'false positives' -- it seems to
detect a change in a component which hasn't changed, or which hasn't been
affected by the code in your PR.

If you've looked at the Chromatic diffs and can't see any connection to your
code, please reach out to a member of the Web Experiences team, who will be able
to advise. It would also be helpful to add the false positive to our
[ongoing log of false positives](https://docs.google.com/spreadsheets/d/1FvItNTMFXIpI4rCrZ4mQ0CRouT06sSVro168f6oKPm4/edit?usp=drive_open&ouid=117150399571694275917#gid=0).
-->

mxdvl reviewed (approved):
Very clean!

I think a lot of the 19 commits can be removed by rebasing this branch?

mxdvl reviewed (commented):
Just a thought: we could use the `gap` property on the flex parent and simplay `display: none` on the input if `signedInUserEmail` is truthy…

lindseydew commented:
> Very clean!
> 
> I think a lot of the 19 commits can be removed by rebasing this branch?

Yep, we can squash those

lindseydew commented:
> Just a thought: we could use the `gap` property on the flex parent and simplay `display: none` on the input if `signedInUserEmail` is truthy…

Yeah we took this approach initially. The problem was that the input box was displayed by default and then removed if this email address was there after the async call and it looked jumpy on the page

lindseydew commented:
> > Just a thought: we could use the `gap` property on the flex parent and simplay `display: none` on the input if `signedInUserEmail` is truthy…
> 
> ~Yeah we took this approach initially. The problem was that the input box was displayed by default and then removed if this email address was there after the async call and it looked jumpy on the page~

Think I'd misunderstood. 

I added a gap to the parent element, and removed the conditional margin-right on the input box. However, this does still introduce a space before the button in the case where the input box is hidden. I think because the button is still showing in the second flex position? Please let me know if I am missing something though

```
const flexParentStyles = css`
	display: flex;
	flex-direction: row;
	align-items: flex-start;
	flex-wrap: wrap;
	gap: ${space[3]}px;
`;
```

mxdvl commented:
> I added a gap to the parent element, and removed the conditional margin-right on the input box. However, this does still introduce a space before the button in the case where the input box is hidden. I think because the button is still showing in the second flex position?

Yes, for this approach to work we would need to ensure that the text input containter `div`  is set to `display: none;` whenever there’s a user email, rather than setting its `flex-basis` to 0.

mxdvl commented on dotcom-rendering/src/components/SecureSignup.importable.tsx:
```diff
@@ -343,41 +344,30 @@ export const SecureSignup = ({ newsletterId, successDescription }: Props) => {
-						`,
-					]}
+					cssOverrides={[labelStyles]}
+					style={
+						signedInUserEmail
+							? { display: 'none' }
+							: { display: 'inline-block' }
+					}
```
You can also set the value to `undefined` to let it just be the default value: 

```suggestion
					style={{
						display:
						signedInUserEmail ? 'none' : undefined
					}}
```

mxdvl reviewed (approved):
Great! Even more approved!
